core/ordersync/ordersync.go

```diff
@@ -159,13 +162,14 @@ type Subprotocol interface {
 // them. New expects an array of subprotocols which the service will support, in the
 // order of preference. The service will automatically pick the most preferred protocol
 // that is supported by both peers for each request/response.
-func New(ctx context.Context, node *p2p.Node, subprotocols []Subprotocol) *Service {
+func New(ctx context.Context, orderfilter *orderfilter.Filter, node *p2p.Node, subprotocols []Subprotocol) *Service {
```

Comment:
If possible, I would prefer not to pass in `orderfilter` here. Future subprotocols might not use filters in the same way or at all.

For example, in the very near future we could implement a subprotocol for getting orders for a specific market. A filter is an inefficient way to do that because it requires getting _all_ orders from the database in memory and checking if each one matches the filter. It would be more optimal to pass in the market asset data because that can be checked directly against the database.

Can we pass in the filter to each subprotocol that uses it instead? Currently that would be both `FilteredPaginationSubProtocolV0` and `FilteredPaginationSubProtocolV1`.